sdk/storage/Azure.Storage.Blobs/src/BlobBaseClient.cs

```diff
@@ -5792,6 +5793,75 @@ protected internal virtual BlobContainerClient GetParentBlobContainerClientCore(
             return _parentBlobContainerClient;
         }
         #endregion
+        #region PartitionedDownloader
+        internal PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult> GetPartitionedDownloader(
+            StorageTransferOptions transferOptions,
+            string operationName = null)
+            => new PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>(
+                GetPartitionedDownloaderBehaviors(this),
+                SanitizePartitionedDownloaderOptions(transferOptions),
+                operationName);
+
+        internal static PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>.Behaviors GetPartitionedDownloaderBehaviors(BlobBaseClient client)
+        {
+            return new PartitionedDownloader<BlobRequestConditions, BlobDownloadStreamingResult>.Behaviors
+            {
+                SingleDownload = async (range, args, rangeGetContentHash, async, cancellationToken, etag)
+                    => await client.DownloadStreamingInternal(
+                        range,
+                        args,
+                        rangeGetContentHash,
+                        $"{nameof(BlobBaseClient)}.{nameof(DownloadTo)}",
+                        async,
+                        cancellationToken).ConfigureAwait(false),
+                ModifyConditions = (args, etag)
+                    => args?.WithIfMatch(etag) ?? new BlobRequestConditions { IfMatch = etag },
+                Scope = operationName => client.ClientConfiguration.ClientDiagnostics.CreateScope(operationName
+                        ?? $"{nameof(Azure)}.{nameof(Storage)}.{nameof(Blobs)}.{nameof(BlobBaseClient)}.{nameof(BlobBaseClient.DownloadTo)}")
+            };
+        }
+
+        internal static StorageTransferOptions SanitizePartitionedDownloaderOptions(
+            StorageTransferOptions transferOptions)
```

Comment:
nit: I'm not sure "sanitize" is right term. `PopulateDefaults` or `ApplyDefaults` perhaps ?
btw. consider extracting this from here, `BlobExtensions` might be good place. 
Also this could be extension method, so we could write `transferOptions.ApplyDefaults()`.